exercises/anagram/src/anagram.h

```diff
@@ -1,13 +1,33 @@
 #ifndef ANAGRAM_H
 #define ANAGRAM_H
 
-#define MAX_STR_LEN 20
+enum anagramStatus {
+   UNCHECKED = -1,
+   NOT_ANAGRAM,
+   IS_ANAGRAM
+};
 
-struct vector {
-   char (*vec)[MAX_STR_LEN];
-   int size;
+struct candidate {
+   enum anagramStatus isAnagram;
+   char *candidate;
 };
 
-struct vector anagrams_for(char *, struct vector);
+struct candidates {
+   struct candidate *candidate;
+   int size;
+};
+/**
+ * @description - determines if any of the words in candidate are anagrams
+ *                for word.  Word buffer and candidate structures may be modified.
+ *
+ * @param - word, buffer containing a word used to look for an anagram in  candidates.
+ *          may be modified before function returns.
+ *
+ * @param - candidates, list of potential anagrams as an array of struct candidate.
+ *          Words in candidate structure may be modifies before function returns.
+ *          candidate.isAnagram will be set to IS_ANAGRAM if the candidate is an
+ *          anagram for word.
+ */
```

Comment:
Actually, I don't think we've really done these big comment blocks before. Does anybody have any thoughts on this? I'm tempted to say that the current signature is pretty clear and that this could be omitted in the name of consistency.

Reply:
Between the signature and the tests the usage is pretty clear IMO.